### ISSUE TITLE
Type '"fill"' is not assignable to type '"fixed" | "intrinsic" | "responsive" | undefined'.

### ISSUE THREAD

**Issue by deadcoder0904** [vercel/next.js#25440]
### What version of Next.js are you using?

10.2.3

### What version of Node.js are you using?

16.1.0

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

next start

### Describe the Bug

I am using a simple `next/image` component:

```js
import Image, { ImageProps } from 'next/image'

export const Img = ({ className = '', ...props }: ImageProps) => (
	<div className="unset-img full-bleed">
		<Image className={`${className} custom-img`} layout="fill" {...props} />
	</div>
)
```

I get red-squiggly lines on	`layout` saying:

> Type '"fill"' is not assignable to type '"fixed" | "intrinsic" | "responsive" | undefined'.

### Expected Behavior

It shouldn't show a TS error.

### To Reproduce

Clone [this commit](https://github.com/deadcoder0904/better-code-blocks/commit/057fcce49fbdab9620840d775273c152745faede), install the dependencies & run the app. VSCode should yell.

The error is in [this file](https://github.com/deadcoder0904/better-code-blocks/blob/057fcce49fbdab9620840d775273c152745faede/src/components/mdx/Img.tsx). See `layout="fill` there.

**Comment by brandonchinn178** [comment]
This might be slightly different from #25344. What happens if someone passes in `layout` to `props`? Or what if someone passes in `width`/`height` to `props` (`layout=fill` is only valid if width/height not provided)?

**Comment by deadcoder0904** [comment]
Cool, so it's different. Is there an easy fix to this?

**Comment by brandonchinn178** [comment]
I think you need to contrain your type more. ImageProps in general doesn't work; for one, the caller shouldnt pass layout, so it's at least Omit<ImageProps, 'layout'>. You'll need to do further changes to handle the other cases I mentioned

**Comment by deadcoder0904** [comment]
@brandonchinn178 not sure `ImageProps` has to do anything with it as I removed it from beside the `props` & it still threw an error.

**Comment by brandonchinn178** [comment]
That could be because it then types `props` as unknown and then typescript squashes the type union and tries to find a version of layout that fits all the possible alternatives in ImageProps (since the Image component is still typed as ImageProps, even if you remove the type from your component)

**Comment by kb1995** [comment]
I found this issue because I'm also experiencing the same problem when I try to pass `preload="true"` prop to the <Image> component. 

The red-squiggly line disappears if I remove the `preload` prop.

```
        <Image
          layout="fill"
          preload="true"
          src={image.url}
          alt={image.alt}
          objectFit="cover"
        />
        ```

**Comment by ycscholes** [comment]
> @brandonchinn178 not sure `ImageProps` has to do anything with it as I removed it from beside the `props` & it still threw an error.

Try to use the type below to replace the default ImageProps
```typescript
type FillImageProps = Omit<ImageProps, 'layout' | 'height' | 'width'>;
```

In your case, you can do like this

```typescript
import Image, { ImageProps } from 'next/image'

type FillImageProps = Omit<ImageProps, 'layout' | 'height' | 'width'>;

export const Img = ({ className = '', ...props }: FillImageProps) => (
	<div className="unset-img full-bleed">
		<Image className={`${className} custom-img`} layout="fill" {...props} />
	</div>
)

```

**Comment by deadcoder0904** [comment]
@ycscholes thanks for the answer but nope, it still gives the same error.

> Type '"fill"' is not assignable to type '"fixed" | "intrinsic" | "responsive" | undefined'.ts(2322)

<img width="781" alt="TS Next.js Error" src="https://user-images.githubusercontent.com/16436270/120184876-1d8d8f00-c22f-11eb-9b87-825ca0af497e.png">

**Comment by deadcoder0904** [comment]
For now, I've found a tiny solution that doesn't spread rest parameters but it works for me:

```tsx
import Image, { ImageProps } from 'next/image'

export const Img = ({ className = '', src }: ImageProps) => (
	<div className="unset-img full-bleed">
		<Image className={`${className} custom-img`} src={src} layout="fill" />
	</div>
)
```

Notice, I need to pass individual props. Ideally, I'd like a solution where I can pass rest params to `Image` component but that component is a little nightmare to work with haha :)

**Comment by ycscholes** [comment]
> @ycscholes thanks for the answer but nope, it still gives the same error.
> 
> > Type '"fill"' is not assignable to type '"fixed" | "intrinsic" | "responsive" | undefined'.ts(2322)
> 
> <img alt="TS Next.js Error" width="781" src="https://user-images.githubusercontent.com/16436270/120184876-1d8d8f00-c22f-11eb-9b87-825ca0af497e.png">

It's my mistake, I use 10.2.0 to test and it works. When i upgrade to 10.2.3, it actually failed.
If you dont need 'placeholder' and 'blurDataURL' attribute, the code below works for me. 
But, it's just a temporary solution, we still need an official solution.

```typescript
type FillImageProps = Omit<
  ImageProps,
  'layout' | 'height' | 'width' | 'placeholder' | 'blurDataURL'
>;

export const Img = ({ className = '', ...props }: FillImageProps) => (
  <div className="unset-img full-bleed">
    <Image className={`${className} custom-img`} layout="fill" {...props} />
  </div>
);
```

**Comment by deadcoder0904** [comment]
@ycscholes Can confirm that works. Maybe `layout` shouldn't have `width`, `height`, `placeholder`, & `blurDataURL`. But not sure so I'll keep this open :)

**Comment by buzzysin** [comment]
I managed to trick the typechecker to not cry when I want to use `layout: "fill"`. I've actually waited for a fix for this since the release of `next/image`, but I've been using this bizarre hackaround:

```tsx
export const FrameBase = (props: ImageProps) => (
  <>
    {
      "layout" in props 
        ? <Image {...props} /> 
        : <Image {...props} />
    }
  </>
);
```

VSCode Hint:

![image](https://user-images.githubusercontent.com/41271986/120893938-ab8dbf00-c60d-11eb-975c-604b96d8cfff.png)

I'm not aware if this does anything, and I'm ***certainly*** no expert, but I think it has something to do with the assertion of the "layout" prop doing some magic with the type guards. Who knows why `Omit` doesn't work the same way? ¯\\_(ツ)\_/¯
